File: Assets/Editor/SceneAPI/Modules/RemoveComponentModule.cs
using System;
using System.Net;
using Newtonsoft.Json;
using UnityEngine;

namespace SceneAPI.Modules
{
    public static class RemoveComponentModule
    {
        public static string Execute(HttpListenerContext context)
        {
            try
            {
                string requestBody = GetRequestBody(context);
                var data = JsonConvert.DeserializeObject<dynamic>(requestBody);
                
                string objectPath = data?.path;
                string componentType = data?.componentType;
                
                if (string.IsNullOrEmpty(objectPath) || string.IsNullOrEmpty(componentType))
                {
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = false, 
                        error = "Object path and component type are required" 
                    });
                }

                GameObject obj = GameObjectUtilities.FindGameObjectByPath(objectPath);
                if (obj == null)
                {
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = false, 
                        error = "Object not found" 
                    });
                }

                Component component = obj.GetComponent(componentType);
                if (component == null)
                {
                    return JsonConvert.SerializeObject(new 
                    { 
                        success = false, 
                        error = $"Component {componentType} not found on object" 
                    });
                }

                UnityEngine.Object.DestroyImmediate(component);
                
                return JsonConvert.SerializeObject(new 
                { 
                    success = true, 
                    message = $"Component {componentType} removed from {objectPath}" 
                });
            }
            catch (Exception ex)
            {
                return JsonConvert.SerializeObject(new 
                { 
                    success = false, 
                    error = $"Error removing component: {ex.Message}" 
                });
            }
        }

        private static string GetRequestBody(HttpListenerContext context)
        {
            using (var reader = new System.IO.StreamReader(context.Request.InputStream))
            {
                return reader.ReadToEnd();
            }
        }
    }
}

File: Assets/PythonMCP/modules/remove_component_module.py
import requests
import json
from typing import Dict

class RemoveComponentModule:
    def __init__(self, base_url: str):
        self.base_url = base_url
    
    def execute(self, object_path: str, component_type: str) -> Dict:
        """Удаляет компонент с объекта"""
        try:
            if not all([object_path, component_type]):
                return {
                    "success": False,
                    "action": "remove_component",
                    "error": "object_path and component_type are required"
                }
            
            response = requests.delete(
                f"{self.base_url}/objects/components/remove", 
                json={"path": object_path, "componentType": component_type}
            )
            response.raise_for_status()
            remove_result = response.json()
            
            return {
                "success": remove_result.get("success", False),
                "action": "remove_component",
                "data": remove_result if remove_result.get("success") else None,
                "error": remove_result.get("error")
            }
            
        except requests.exceptions.RequestException as e:
            return {
                "success": False,
                "action": "remove_component",
                "error": f"Request error: {str(e)}"
            }
        except json.JSONDecodeError as e:
            return {
                "success": False,
                "action": "remove_component",
                "error": f"JSON decode error: {str(e)}"
            }